src/CommandingOfficers/DeployableCommander.java

```diff
@@ -0,0 +1,263 @@
+package CommandingOfficers;
+
+import Engine.GameScenario;
+
+import java.util.ArrayList;
+import CommandingOfficers.CommanderInfo.InfoPage;
+import Engine.GameEvents.GameEvent;
+import Engine.GameEvents.GameEventListener;
+import Engine.GameEvents.GameEventQueue;
+import Engine.UnitActionLifecycles.JoinLifecycle.JoinEvent;
+import Engine.Army;
+import Engine.GameAction;
+import Engine.GameActionSet;
+import Engine.GamePath;
+import Engine.UnitActionFactory;
+import Engine.XYCoord;
+import Terrain.GameMap;
+import Terrain.MapLocation;
+import Terrain.MapMaster;
+import Terrain.TerrainType;
+import UI.MapView;
+import UI.Art.Animation.GameAnimation;
+import Units.Unit;
+import Units.UnitContext;
+import Units.UnitModel;
+
+public abstract class DeployableCommander extends Commander
+{
+  private static final long serialVersionUID = 1L;
+  public static final InfoPage COU_MECHANICS_BLURB = new InfoPage(
+            "CO Unit mechanics:\n"
+          + "Deploy your CO Unit via a special action done in place (does not end turn).\n"
+          + "You can't deploy a CO Unit that has already been active during your turn (i.e. must wait a turn after deleting it/dying to a counter).\n");
+  public ArrayList<Unit> COUs = new ArrayList<Unit>();
+  public ArrayList<Unit> COUsLost = new ArrayList<Unit>();
+  public final DeployCOUFactory deployAction; // Has to be one per CO, so units get the same instance over multiple action searches (mostly for the AI)
+  /** The number of COUs you can have active at once */
+  public abstract int getCOUCount();
+  public void onCOULost(Unit minion) {};
+  public char getCOUMark() {return 'C';};
+  public int deployCostPercent = 0;
+  /** If the unit type matches any flag in this mask, it can be my COU */
+  public long canDeployMask = Long.MAX_VALUE;
+  public final boolean canDeployOn(UnitModel type) {return type.isAny(canDeployMask);};
+
+  public boolean resetCOUsEveryTurn = false;
+
+  /** Can I deploy my COU here? Ignored if resetCOUsEveryTurn==true */
+  protected boolean eligibleDeployLocation(Unit actor, MapLocation loc)
+  {
+    if( loc.getOwner() != actor.CO )
+      return false; // No COing on neutral factories
```

Comment:
So if `resetCOUsEveryTurn==true`, then we are allowed to CO on neutral factories?

Reply:
If `resetCOUsEveryTurn==True`, you can put your "COU" on any unit you want.
Maybe "COU anywhere" shouldn't be tied to "COU every turn", since I guess you could pull off some funny designs with that... but all the current usecases did it that way.

Limiting your COUs to one turn away from your production seems pretty bad, lol

Reply:
Fair point. Are there COs that have to redeploy themselves every turn? That seems... odd.

Reply:
That was the initial draft of the Tabitha conversion, and still is Lin's.
My idea is basically: "Lin has a zone of 1, so she should be more flexible than Tabitha. But having a real zone mechanic is too much... so we can just sort of assume Lin has her zone where it needs to be to boost one critical unit each turn".
Whether that's a good conversion or even actually simpler than setting up zones for real? Good question :v